snuba_sdk/query_validation.py

```diff
@@ -0,0 +1,224 @@
+from __future__ import annotations
```

Comment:
The match validation in here was all copied out of the query_visitors validation and restructured slightly, except for checking the data model for columns in `validate_entity`